### PR TITLE
perf: Antd icons tree shaking

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY
Previously we were importing the whole @ant-design/icons library, which contained over 800 icons. We only use 40 of those.
Implementing tree-shakable import allowed to shrink the chunk size by over 5MB.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:

Screenshot 2024-07-31 at 15 32 27

After:

Screenshot 2024-07-31 at 15 34 58

### TESTING INSTRUCTIONS
Everything should work like before

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by geido** [comment]
While I understand the need for this, the idea here was that you could use any Antd icon without having to specify it again. Not sure if there is a way around that to keep that functionality available.

**Comment by michael-s-molina** [comment]
Agree with @kgabryje 👍🏼

File: superset-frontend/src/components/Icons/AntdEnhanced.tsx [hunk 1]
```diff
@@ -17,10 +17,94 @@
  * under the License.
  */
 
-import * as AntdIcons from '@ant-design/icons/lib/icons';
+import {
```
Comment (mistercrunch) [inline superset-frontend/src/components/Icons/AntdEnhanced.tsx:20]: would be good to add a comment
```
// Targeted import (as opposed to `import * as antdIcons`) is important here for proper tree-shaking
```

File: superset-frontend/src/components/Icons/AntdEnhanced.tsx [hunk 2]
```diff
@@ -17,10 +17,94 @@
...
+  SettingOutlined,
+  StopOutlined,
+  TagsOutlined,
+  UpOutlined,
+  UserOutlined,
+} from '@ant-design/icons';
 import { StyledIcon } from './Icon';
 import IconType from './IconType';
 
+const AntdIcons = {
```
Comment (mistercrunch) [inline superset-frontend/src/components/Icons/AntdEnhanced.tsx:65]: I looked for ways to keep things DRY here and not have to repeat icon names twice, but didn't find any way. Too bad ECMAScript doesn't support something like `import {a, b, c} as container from ....`

**Review by mistercrunch — APPROVED** [review]
LGTM - though would be nice to add this comment as someone reading the code might think "why did they not just `import *`!?"

**Review by rusackas — APPROVED** [review]
LG2M2!

File: superset-frontend/src/components/Icons/AntdEnhanced.tsx (hunk 1 above)
Comment (mistercrunch) [reply superset-frontend/src/components/Icons/AntdEnhanced.tsx:21]: ok I added a commit with the comment - first time using that GH feature...

Comment (rusackas) [reply superset-frontend/src/components/Icons/AntdEnhanced.tsx:21]: > first time using that GH feature...

I'm in love with it... also great for fixing little linting issues on PRs and such.

Comment (kgabryje) [reply superset-frontend/src/components/Icons/AntdEnhanced.tsx:21]: oh nice! never used it myself

**Pull request merged**